protocols/zcash-frost/src/protocols/keygen.rs

```diff
@@ -0,0 +1,585 @@
+    logger: DebugLogger,
+    account_id: AccountId,
+}
+
+pub async fn create_protocol<B, BE, KBE, C, N>(
+    account_id: AccountId,
+    client: JobsClient<B, BE, C>,
+    network: N,
+    logger: DebugLogger,
+    key_store: ECDSAKeyStore<KBE>,
+) -> ZcashFrostKeygenProtocol<B, BE, KBE, C, N>
+where
+    B: Block,
+    BE: Backend<B>,
+    C: ClientWithApi<B, BE>,
+    KBE: KeystoreBackend,
+    N: Network,
+    <C as ProvideRuntimeApi<B>>::Api: JobsApi<B, AccountId>,
+{
+    ZcashFrostKeygenProtocol {
+        client,
+        network,
+        key_store,
+        logger,
+        account_id,
+    }
+}
+
+#[async_trait]
+impl<
+        B: Block,
+        BE: Backend<B> + 'static,
+        C: ClientWithApi<B, BE>,
+        KBE: KeystoreBackend,
+        N: Network,
+    > GadgetProtocol<B, BE, C> for ZcashFrostKeygenProtocol<B, BE, KBE, C, N>
+where
+    <C as ProvideRuntimeApi<B>>::Api: JobsApi<B, AccountId>,
+{
+    fn name(&self) -> String {
+        "zcash-frost-keygen".to_string()
+    }
+
+    async fn create_next_job(
+        &self,
+        job: JobInitMetadata<B>,
+    ) -> Result<<Self as AsyncProtocol>::AdditionalParams, gadget_common::Error> {
+        let job_id = job.job_id;
+        let role_type = job.job_type.get_role_type();
+
+        // We can safely make this assumption because we are only creating jobs for phase one
+        let JobType::DKGTSSPhaseOne(p1_job) = job.job_type else {
+            panic!("Should be valid type")
+        };
+
+        let participants = p1_job.participants;
+        let threshold = p1_job.threshold;
+
+        let user_id_to_account_id_mapping = Arc::new(
+            participants
+                .clone()
+                .into_iter()
+                .enumerate()
+                .map(|r| (r.0 as UserID, r.1))
+                .collect(),
+        );
+
+        let params = ZcashFrostKeygenExtraParams {
+            i: participants
+                .iter()
+                .position(|p| p == &self.account_id)
+                .expect("Should exist") as u16,
+            t: threshold as u16,
+            n: participants.len() as u16,
+            role_type,
+            job_id,
+            user_id_to_account_id_mapping,
+        };
+
+        Ok(params)
+    }
+
+    async fn process_block_import_notification(
+        &self,
+        _notification: BlockImportNotification<B>,
+        _job_manager: &ProtocolWorkManager<WorkManager>,
+    ) -> Result<(), gadget_common::Error> {
+        Ok(())
+    }
+
+    async fn process_error(
+        &self,
+        error: gadget_common::Error,
+        _job_manager: &ProtocolWorkManager<WorkManager>,
+    ) {
+        log::error!(target: "zcash_frost", "Error: {error:?}");
+    }
+
+    fn account_id(&self) -> &AccountId {
+        &self.account_id
+    }
+
+    fn role_filter(&self, role: RoleType) -> bool {
+        matches!(
+            role,
+            RoleType::Tss(ThresholdSignatureRoleType::DfnsCGGMP21Secp256k1)
+        )
```

Comment:
Fix the role type here:
```suggestion
        matches!(
            role,
            RoleType::Tss(ThresholdSignatureRoleType::ZcashFrostSr25519)
            | RoleType::Tss(ThresholdSignatureRoleType::ZcashFrostP256)
            | RoleType::Tss(ThresholdSignatureRoleType::ZcashFrostSecp256k1)
            | RoleType::Tss(ThresholdSignatureRoleType::ZcashFrostRistretto255)
            | RoleType::Tss(ThresholdSignatureRoleType::ZcashFrostEd25519)
        )
```

Reply:
That was the blocker @drewstone for your tests not running correctly.